nuxeo-management-rest-api/src/test/java/org/nuxeo/rest/management/TestRenditionsObject.java

```diff
@@ -0,0 +1,222 @@
+    protected ThumbnailService thumbnailService;
+
+    @Test
+    public void testRendtionsPostPicturesRecompute() throws IOException {
+        // test the picture views are computed correctly and trashing them
+        DocumentRef docRef = beforePictureViewsRecompute();
+
+        // generating new picture views
+        runBulkAction("/pictures/recompute");
+
+        // checking the picture views are recomputed in the session
+        afterPicturesRecompute(docRef);
+    }
+
+    @Test
+    @Deploy("org.nuxeo.ecm.platform.thumbnail")
+    public void testRendtionsPostThumbnailsRecomputeBulkStatus() throws IOException {
+        // check the original thumbnail is computed and trashing it
+        DocumentRef docRef = beforeThumbnailsRecompute();
+
+        // generating a new thumbnail
+        runBulkAction("/thumbnails/recompute");
+
+        // checking the thumbnail is recomputed in the session
+        afterThumbnailsRecompute(docRef);
+    }
+
+    protected JsonNode validateBulkStatus(CloseableClientResponse response) throws IOException {
+        assertEquals(HttpServletResponse.SC_OK, response.getStatus());
+        JsonNode node = mapper.readTree(response.getEntityInputStream());
+        JsonNode value = node.get(ENTITY_TYPE);
+        assertEquals(STATUS_ENTITY_TYPE, value.asText());
+        assertNotNull(node.get("state"));
+        assertNotNull(node.get("processed"));
+        assertNotNull(node.get("error"));
+        assertNotNull(node.get("errorCount"));
+        assertNotNull(node.get("total"));
+        assertNotNull(node.get("action"));
+        assertNotNull(node.get("username"));
+        assertNotNull(node.get("submitted"));
+        assertNotNull(node.get("scrollStart"));
+        assertNotNull(node.get("scrollEnd"));
+        assertNotNull(node.get("processingStart"));
+        assertNotNull(node.get("processingEnd"));
+        assertNotNull(node.get("completed"));
+        assertNotNull(node.get("processingMillis"));
+        return node;
+    }
+
+    protected void runBulkAction(String extraPath) throws IOException {
+        String fullPath = PATH + extraPath;
+        try (CloseableClientResponse response = httpClientRule.post(fullPath, null)) {
+            JsonNode result = validateBulkStatus(response);
+            JsonAssert jsonAssert = JsonAssert.on(result.toString());
+            jsonAssert.get("state").equals(State.SCHEDULED);
+
+        }
+        // waiting for the asynchronous BAF task to finish
+        txFeature.nextTransaction();
+
+        // checking the picture views are recomputed in the BulkStatus
+        try (CloseableClientResponse response = httpClientRule.get(fullPath)) {
+            JsonNode result = validateBulkStatus(response);
+            JsonAssert jsonAssert = JsonAssert.on(result.toString());
+            jsonAssert.get(ENTITY_TYPE).equals(STATUS_ENTITY_TYPE);
+            jsonAssert.get("state").equals(State.COMPLETED);
+            jsonAssert.get("completed").notNull();
```

Comment:
This is already handled by validateBulkStatus.
Or maybe here in the case of a completed computation we you want to check something more on this "completed" field ? That it actually holds a date in a string format;

Reply:
I'm checking the state is completed after being scheduled (when it fails it give me unknown usually)

Since `BulkStatus.getCompletedTime()` returns an `Instant` I propose to check the date with `Instant.parse(result.get("completed").asText())`

Reply:
You could but might be tricky to have the ritgh sate. Maybe just check that the node value is not null (here you only check that the "completed" node exists).